rust/gui-client/src-tauri/src/controller.rs

```diff
@@ -622,7 +652,41 @@ impl<I: GuiIntegration> Controller<I> {
         Ok(ControlFlow::Continue(()))
     }
 
-    async fn handle_connect_result(&mut self, result: Result<(), ipc::Error>) -> Result<()> {
+    async fn handle_gui_ipc_msg(
+        &mut self,
+        maybe_msg: Option<Result<gui::ClientMsg>>,
+    ) -> Result<()> {
+        let client_msg = maybe_msg
+            .context("No message received")?
+            .context("Failed to read message")?;
+
+        match client_msg {
+            gui::ClientMsg::Deeplink(url) => match self.handle_deep_link(&url).await {
+                Ok(()) => {}
+                Err(error)
+                    if error
+                        .root_cause()
+                        .downcast_ref::<auth::Error>()
+                        .is_some_and(|e| matches!(e, auth::Error::NoInflightRequest)) =>
+                {
+                    tracing::debug!("Ignoring deep-link; no local state");
+                }
+                Err(error) => {
+                    tracing::error!("`handle_deep_link` failed: {error:#}");
+                }
+            },
+            gui::ClientMsg::NewInstance => {
+                tracing::debug!("A new instance of Firezone has been launched")
```

Comment:
@jamilbk Here is where we will be able to react to additional instances of Firezone being launched.